zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -16,231 +16,288 @@ enum class Status {
     NOT_APPLICABLE,
 }
 
-// Validator if Kleinverbruik or Grootverbruik data
-val validateKleinverbruikOrGrootverbruik = Validator { survey: Survey ->
-    val gridConnection = survey.getSingleGridConnection()
+class SurveyValidator {
+    fun validate(survey: Survey): List<ValidationResult> {
+        val results = mutableListOf<ValidationResult>()
 
-    when (gridConnection.electricity.kleinverbruikOrGrootverbruik) {
-        KleinverbruikOrGrootverbruik.GROOTVERBRUIK -> {
-            gridConnection.electricity.grootverbruik?.let {
-                ValidationResult(Status.VALID, "Large consumption data has been provided")
-            } ?: ValidationResult(Status.INVALID, "Large consumption data is not provided")
+        survey.addresses.forEach { address ->
+            results.addAll(AddressValidator().validate(address))
         }
-        KleinverbruikOrGrootverbruik.KLEINVERBRUIK -> {
-            gridConnection.electricity.kleinverbruik?.let {
-                ValidationResult(Status.VALID, "Small consumption data has been provided")
-            } ?: ValidationResult(Status.INVALID, "Small consumption data is not provided")
-        }
-        else -> ValidationResult(Status.INVALID, "Neither small nor large consumption type is provided")
-    }
-}
 
-// Validator GROOTVERBRUIK has required values
-val validateGrootverbruikCapacities = Validator { survey: Survey ->
-    val gridConnection = survey.getSingleGridConnection()
-    if (gridConnection.electricity.kleinverbruikOrGrootverbruik == KleinverbruikOrGrootverbruik.GROOTVERBRUIK) {
-        val grootverbruik = gridConnection.electricity.grootverbruik
-        grootverbruik?.contractedConnectionDeliveryCapacity_kW ?: ValidationResult(Status.MISSING_DATA, "Connection delivery capacity is not provided for large consumption")
-        grootverbruik?.physicalCapacityKw ?: ValidationResult(Status.MISSING_DATA, "Physical Capacity Kw is not provided for large consumption")
-        grootverbruik?.contractedConnectionFeedInCapacity_kW ?: ValidationResult(Status.MISSING_DATA, "Connection delivery capacity is not provided for large consumption")
+        return results
     }
-    ValidationResult(Status.NOT_APPLICABLE, "large consumption values has been reviewed")
 }
 
-// Validator for contracted delivery capacity <= physical capacity
-val validateContractedCapacity = Validator { survey: Survey ->
-    val gridConnection = survey.getSingleGridConnection()
-    val contractedCapacity = gridConnection.electricity.getContractedConnectionCapacityKw()
-    val physicalCapacity = gridConnection.electricity.getPhysicalConnectionCapacityKw()
+class AddressValidator {
```

Comment:
Class isn't necessary if it's just one function

Reply:
Good point